pkg/kernel/probe.go

```diff
@@ -136,11 +166,7 @@ func (p *Probe) Run(ctx context.Context, cb DataCallback) error {
 	}
 
 	if !p.initialized {
-		// don't rely on end-user to call init (as it could be sourced from
-		// eventreader)
-		if err := p.InitProbe(); err != nil {
-			return err
-		}
```

Comment:
This change was intentional - this type of helper just attempted to cover for incorrect API usage - the idea here is to push the user into using the API correctly.